core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -2854,6 +2849,11 @@ castUnsafe (MutArray contents start end bound) =
 asBytes :: MutArray a -> MutArray Word8
 asBytes = castUnsafe
 
+-- | Cast an @MutArray Word8@ into an @MutArray a@.
+--
+fromBytes :: MutArray Word8 -> MutArray a
+fromBytes = castUnsafe
+
```

Comment:
why introduce this API? `castUnsafe` can be used directly.

Reply:
If I use `castUnsafe` I'll have to type it manually. Just like `asBytes`, `fromBytes` makes sense to have.
Especially since we are removing `newAsBytes`